Assets/Scripts/GameManager.cs

```diff
@@ -96,6 +98,42 @@ private void Start()
         }
     }
 
+    private void Update()
+    {
+        if (PhotonNetwork.IsMasterClient && !gameFinished)
+        {
+            GameObject[] players = GameObject.FindGameObjectsWithTag("Player");
+
+            if (GameObject.FindGameObjectWithTag("Boss").GetComponent<BossManagerCore>().GetHealth() <= 0)
+            {
+                gameFinished = true;
+
+                // Destroy all players
+                foreach (GameObject player in players)
```

Comment:
It isn't necessary to loop through the players, since DestroyAll() destroys everything in the scene, for all clients.